src/fu-util.c

```diff
@@ -1304,11 +1313,50 @@ fu_util_unlock (FuUtilPrivate *priv, gchar **values, GError **error)
 	return fwupd_client_unlock (priv->client, values[0], NULL, error);
 }
 
+static gboolean
+fu_util_perhaps_refresh_remotes (FuUtilPrivate *priv, GError **error)
+{
+	g_autoptr(GPtrArray) remotes = NULL;
+	guint64 age_oldest = 0;
+	const guint64 age_limit_days = 30;
+
+	/* get the age of the oldest enabled remotes */
+	remotes = fwupd_client_get_remotes (priv->client, NULL, error);
+	if (remotes == NULL)
+		return FALSE;
+	for (guint i = 0; i < remotes->len; i++) {
+		FwupdRemote *remote = g_ptr_array_index (remotes, i);
+		if (!fwupd_remote_get_enabled (remote))
+			continue;
+		if (fwupd_remote_get_kind (remote) != FWUPD_REMOTE_KIND_DOWNLOAD)
+			continue;
+		if (fwupd_remote_get_age (remote) > age_oldest)
+			age_oldest = fwupd_remote_get_age (remote);
+	}
+
+	/* ask the user */
+	if (age_oldest < 60 * 60 * 24 * age_limit_days)
+		return TRUE;
+	if ((priv->flags & FWUPD_INSTALL_FLAG_FORCE) == 0) {
+		g_printerr ("\n******************\n\n");
+		g_printerr ("The metadata is old and may not contain the latest updates!\n");
+		g_printerr ("\n******************\n\n");
+		g_printerr ("%s [Y|n]: ", _("Update now? (requires internet connection)"));
```

Comment:
I think this specific line should be guarded by `priv->assume_no` / `priv->assume-yes`.   If either of those are set you don't need to show this message

Reply:
Fixed.